### DISCUSSION TITLE
太字修飾の**の中に括弧の)部分を入れると太字が適用されない

### DISCUSSION THREAD

**Discussion by zizynonno** [increments/qiita-discussions#195] [要望・アイデア・不具合]
**問題**
Qiitaの投稿画面にて、「)**hogehoge」と打つと太字が適用されなくなることを確認。投稿後の表示も太字が適用されていない。

**エビデンス**
下記、Qiitaの投稿画面スクショ画像です。

編集
![スクリーンショット 2022-04-02 0 28 08](https://user-images.githubusercontent.com/34001074/161294564-7b269a22-cc8f-4c3e-9450-992ff93bb2e2.png)

プレビュー
![スクリーンショット 2022-04-02 0 28 31](https://user-images.githubusercontent.com/34001074/161294636-88108dff-a3f8-41c8-8a88-dd19183fcd6d.png)

**Comment by kskwtnk** [comment]
@zizynonno 
ご意見ありがとうございます！

申し訳ありません、こちらGitHub Flavored Markdownの仕様となっていまして…… 🙏 

```
平文 **強調したいテキスト** 平文
```

上記のように、前後にスペースを入れてください 🙏 

詳しくは[ヘルプ](https://help.qiita.com/ja/articles/qiita-markdown#markdown)や[弊社メンバーによる説明記事](https://qiita.com/fussy113/items/b40f60f6803aa2d5cfbc)にも記載してありますので、こちらをご確認いただけると幸いです！

**Comment by tats-u** [comment]
この問題を緩和するremarkプラグインを作成しました。ぜひご検討ください。

https://www.npmjs.com/package/remark-cjk-friendly

紹介記事：https://qiita.com/tats-u/items/f571313793b367d7c326

そこら辺の提案と違い、CommonMark本体に取り込まれることを意図して策定しています。

**Reply by kimkim0814** [reply]
@tats-u
プラグインを作成していただき、ありがとうございます！🙏

Qiita のバックエンドは Ruby（Rails）で構築されており、Markdown の処理には CommonMarker を利用しています。
これにより、バックエンド側で CommonMark および GFM の仕様に準拠した処理を行っています。

一方で、remark は JavaScript 製のライブラリであり、プラグインも JavaScript 上で動作する設計となっています。
そのため、Qiita の Markdown の仕様では現状、remark プラグインを直接利用するのは難しい状況です。

ただし、CommonMark の仕様に提案が取り入れられた場合、CommonMarker にも将来的に反映される可能性があります。
Qiita 側で独自の拡張を検討しますが、すぐに対応するのは難しい点をご理解いただけると幸いです。  

よろしくお願いいたします！🙇‍♂️

**Reply by tats-u** [reply]
CommonMarker内部のComrakの改造がまず必要ですね。失礼しました。
CommonMarkに変更を取り込ませることを第一目標にします。

**Reply by tats-u** [reply]
https://github.com/kivikakk/comrak/issues/569

Comrak作者曰く、拡張として追加するPRはOKとのことです。改造は不要なので、将来CommonMarkerを更新しオプションを追加で渡せば反映される見通しです。

**Reply by acronhub** [reply]
@tats-u 
共有ありがとうとうございます 🙏  
反映されるのを楽しみにしています。

**Reply by tats-u** [reply]
Comrakに変更加えましたので、CommonMarkerの次～次々バージョンあたりで使えるようになると思います。

- https://github.com/kivikakk/comrak/issues/569
- https://github.com/kivikakk/comrak/pull/582

**Reply by tats-u** [reply]
commonmarker 2.3.2以降で利用可能です。
`extensions`内で`cjk_friendly_emphasis: true`を指定してください。

https://github.com/gjtorikian/commonmarker/releases/tag/v2.3.2
https://github.com/gjtorikian/commonmarker?tab=readme-ov-file#extension-options

**Reply by P-SiZK** [reply]
@tats-u 
お知らせいただきありがとうございます！

誠に恐縮ながら、この変更を qiita_marker に取り込むには、現時点ではかなりお時間をいただくことになります :bow:
お手数をおかけしますが、お待ちいただけると幸いです。

**Reply by tats-u** [reply]
https://github.com/increments/qiita_marker の言語構成を見るに、GitHub同様commonmarkerの0系（cmark使用・Comrak不使用）がベースですか？
それでしたら、https://github.com/swiftlang/swift-cmark/pull/78 の差分を取り込むのがよいかと思います。
ある程度見た限りではこちらの意図通り実装されているかと思います。

**Reply by tanaka-asami415** [reply]
@tats-u 
具体的な Pull Requestまでご提示いただき、ありがとうございます！

> GitHub同様commonmarkerの0系（cmark使用・Comrak不使用）がベースですか？

はい！swift-cmarkのPull Requestの詳細を完全には把握できていないのですが、qiita_marker で利用しているcmarkベースの実装として適していそうに思いました。

現在は優先的に進めている開発があり、すぐの対応は難しいのですが、tats-uさんに継続的なご支援をいただいているので、ぜひ取り組みたいと考えております！
お時間をいただき申し訳ありませんが、お待ちいただけると幸いです🙇

**Reply by tats-u** [reply]
https://github.com/vuejs/vitepress/blob/main/CHANGELOG.md#200-alpha12-2025-08-20

参考までにVitepress（Vite・Vue・Mermaidなどのドキュメント生成システム）の次期版でも採用されることになりました。

**Reply by Phroneris** [reply]
2022年の @kskwtnk 氏のコメントの後、こちらのコメントのスレッドで議論が前向きに進んでいるのを見るに、~~このDiscussionがクローズ扱いなのは~~このDiscussionにクローズタグが付いているのは違うように思うのですがいかがですか？

**Reply by tats-u** [reply]
うわ本当じゃん・・・

@P-SiZK @tanaka-asami415 採用予定があるのでしたら、すぐにでも「クローズ」のラベルを外してください。よろしくお願いします。プレイグラウンドも用意しています。

https://tats-u.github.io/markdown-cjk-friendly/

**Reply by tanaka-asami415** [reply]
@tats-u @Phroneris 
大変失礼いたしました。おそらく https://github.com/increments/qiita-discussions/discussions/195#discussioncomment-2504580 の返信の際にクローズして、そのままになっていたようです🙇
「クローズ」ラベルを外しました！

プレイグラウンドもありがとうございます。
引き続きよろしくお願いします。

**Reply by tats-u** [reply]
@tanaka-asami415 手早い対応ありがとうございます。

**Reply by tats-u** [reply]
Swift側完全に停滞しています。
拡張でないのが足引っ張っているのかな？と思って取り消し線同様の拡張にするPRを出したのですがこちらも音沙汰なしです。
https://github.com/swiftlang/swift-cmark/pull/86

**Reply by tats-u** [reply]
https://github.com/tats-u/qiita_marker/pull/1

仕方ないので素のcmark用に書きかけたものベースにCopilotのラストバーゲンセール使って書かせました
現在調整・検証中です

使用：Opus 4.6 + Sonnet 4.6（Cloud Agent経由だとなぜか没収されたはずのOpus 4.6が生きている）
セッションログ：https://github.com/tats-u/qiita_marker/tasks/6f742bb2-40c0-4477-ad46-6fadbe34bec6?session_id=4d74fbbe-5d0a-48c6-b2db-5f504d3c07eb

**Reply by tats-u** [reply]
https://github.com/tats-u/qiita_marker/pull/2

extensionではなくparse_optionでやるバージョン

どちらがよいですか？

**Reply by kimkim0814** [reply]
@tats-u 
ご対応いただきありがとうございます🙏
採用をお約束はできないのですが、それぞれの内容を確認したうえで、どちらで進めるとよいか改めてご連絡させていただきます。

**Reply by tats-u** [reply]
ご検討ありがとうございます。
選択された方を元に調整後PRを出そうと思います。

**Reply by tomoasleep** [reply]
@tats-u 返答お待たせしてしまい申し訳ありません。 また、様々なパターンの patch をご共有いただきありがとうございます。
Qiita としては、以下の方針で取り込みたいと考えています。

* https://github.com/tats-u/qiita_marker/pull/1 の extension 的な実装の方が望ましいと考えています。
  * この方針で導入を行いたいので qiita_marker に Pull Request を送っていただけますと幸いです。 (あるいはこれを参考実装として Qiita 側で qiita_marker への実装を検討します)
* 今後のメンテナンスを考えて、qiita_marker の元の実装ファイルへの変更は最小限になっていて、追加する実装の大部分が別ファイルに分ける形にしたいと考えています。
  * これに関しては Pull Request 上で相談するのがスムーズだと思うので、とりあえず現状のコードのまま Pull Request 送ってくださって構いません。

これについての背景も軽く説明させてください。

qiita_marker では今後ベースとなる Markdown プロセッサの変更を検討しております。なので、今回の patch は、移行先の Markdown プロセッサにも、拡張の移植を行いやすい状態で取り込みたいと考えています。

qiita_marker はご存知の通り CommonMarker v0 (cmark を利用) がベースとなっております。
現状 CommonMarker の現状の実装と乖離が出てしまっているので、以下の検討を行っております。

* CommonMarker の最新版に追従
* https://github.com/socketry/markly などの cmark を利用した別の実装を採用
* あるいは第3の選択肢…

なので、取り込む際は、移植が容易になるように一部修正をお願いすることになるかもしれません。
詳しくは Pull Request 上で相談させて頂ければと思います。よろしくお願いいたします。
